addon/components/qr-scanner.js

```diff
@@ -85,6 +90,26 @@ export default Ember.Component.extend({
     if (timerId) {
       Ember.run.cancel(timerId);
     }
+
+    // Cross browser stream.stop hack from:https://stackoverflow.com/questions/11642926/stop-close-webcam-which-is-opened-by-navigator-getusermedia
+    var MediaStream = window.MediaStream;
+    if (typeof MediaStream === 'undefined' && typeof webkitMediaStream !== 'undefined') {
+        MediaStream = webkitMediaStream;
+    }
+    if (typeof MediaStream !== 'undefined' && !('stop' in MediaStream.prototype)) {
+        MediaStream.prototype.stop = function() {
+            this.getAudioTracks().forEach(function(track) {
+                track.stop();
+            });
+
+            this.getVideoTracks().forEach(function(track) {
+                track.stop();
+            });
+        };
+    }
+
+    // Stop this stream
+    this.get("mediaStream").stop();
```

Comment:
this should be called in `willRemoveElement()` directly as this is not related to cancelling the timer